contentgrid-spring-boot-autoconfigure/src/main/java/com/contentgrid/spring/boot/autoconfigure/data/audit/JpaAuditingAutoConfiguration.java

```diff
@@ -1,18 +1,27 @@
 package com.contentgrid.spring.boot.autoconfigure.data.audit;
 
-import com.contentgrid.spring.data.support.auditing.v1.JpaAuditingConfiguration;
+import com.contentgrid.spring.data.support.auditing.v1.AuditMetadata;
+import com.contentgrid.spring.data.support.auditing.v1.JwtAuditorAware;
+import com.contentgrid.spring.data.support.auditing.v1.UserMetadata;
 import jakarta.persistence.EntityManagerFactory;
 import org.springframework.boot.autoconfigure.AutoConfiguration;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration;
-import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Bean;
+import org.springframework.data.domain.AuditorAware;
+import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
 import org.springframework.security.oauth2.jwt.JwtClaimAccessor;
 
 @AutoConfiguration(after = {HibernateJpaAutoConfiguration.class})
 @ConditionalOnBean(EntityManagerFactory.class)
-@ConditionalOnClass({JpaAuditingConfiguration.class, JwtClaimAccessor.class})
-@Import(JpaAuditingConfiguration.class)
+@ConditionalOnClass(AuditMetadata.class)
```

Comment:
This is not really dependent on the presence of the `AuditMetadata` class, only the `jwtAuditorAware` is dependent on the `UserMetadata` class being present.

Since `UserMetadata` is part of this class signature, you can't put `@ConditionalOnClass()` on the method, as the class would still attempt to get loaded.

Instead, I think 2 separate classes are necessary:
 - one to enable JPA Auditing; unconditionally. That can just be this class
 - one to create the `jwtAuditorAware` bean; conditional on `UserMetadata` and `JwtClaimAccessor` being on the classpath

These don't have to be autoconfigurations or separate files, the class for `jwtAuditorAware` can be a nested `@Configuration` class that is `@Import`ed by the autoconfiguration.